wellcomeml/ml/voting_classifier.py

```diff
@@ -31,28 +34,38 @@ def _get_estimators(self):
 
     def predict(self, X):
         if self.pretrained:
-            check_is_fitted(self, 'estimators')
+            check_is_fitted(self, "estimators")
 
             estimators = self._get_estimators()
 
             if self.voting == "soft":
+                if self.num_agree:
+                    logger.warning("num_agree specified but not needed for soft voting")
                 Y_probs = np.array([est.predict_proba(X) for est in estimators])
                 Y_prob = np.mean(Y_probs, axis=0)
                 if self.multilabel:
                     return np.array(Y_prob > 0.5, dtype=int)
                 else:
                     return np.argmax(Y_prob, axis=1)
             else:  # hard voting
+
                 Y_preds = [est.predict(X) for est in estimators]
+                Y_preds = np.array(Y_preds)
                 if self.multilabel:
-                    Y_preds = np.array(Y_preds)
-                    axis = 0
+                    if self.num_agree:
+                        logger.warning("num_agree specified but not needed for multilabel voting")
+                    return np.apply_along_axis(
+                        lambda x: np.argmax(np.bincount(x)), axis=0, arr=Y_preds
+                    )
                 else:
-                    Y_preds = np.column_stack(Y_preds)
-                    axis = 1
-                return np.apply_along_axis(
-                    lambda x: np.argmax(np.bincount(x)),
-                    axis=axis,
-                    arr=Y_preds)
+                    # If num_agree isn't set then use majority vote
+                    if not self.num_agree:
+                        # So if 4 estimators, >= 3 need to agree
+                        self.num_agree = np.ceil((len(estimators) + 1) / 2)
+
+                    return np.apply_along_axis(
+                        lambda x: int(np.sum(x) >= self.num_agree), axis=0, arr=Y_preds
```

Comment:
I assume this work for binary, does it also work for multi class?

Reply:
For `multilabel` I think something simple like np.array(Y.sum(axis=0) > self.num_agree, dtype=int32) should work.

For `multi class` it is slightly more complicated by here is a two step solution
```
votes = np.apply_along_axis(lambda x: max(np.bincount(x)), axis=0, arr=Y_preds)
max_class = np.apply_along_axis(lambda x: np.argmax(np.bincount(x)), axis=0, arr=Y_preds)
return max_class * (votes > self.num_agree)
```

Reply:
I think that the multi class version above also covers binary

Reply:
these were v useful :) 
In the multiclass option I changed it so that if there is no maximum over the threshold, then it wont default to 0 (as `max_class * (votes > self.num_agree)` does) but rather pick the first value from the classes which were voted on.
e.g. if classes  5, 2, and 3 were voted on twice by different estimators, but `num_agree` was 3, then the output would be class 2.